src/scripts/scripts/zone/coilfang_resevoir/serpent_shrine/boss_fathomlord_karathress.cpp

```diff
@@ -230,23 +230,19 @@ struct boss_fathomlord_karathressAI : public ScriptedAI
     {
         DoScriptText(SAY_DEATH, m_creature);
 
-        if(pInstance)
-            pInstance->SetData(DATA_KARATHRESS_EVENT, DONE);
+        pInstance->SetData(DATA_KARATHRESS_EVENT, SPECIAL);
 
         //support for quest 10944
         m_creature->SummonCreature(SEER_OLUM, OLUM_X, OLUM_Y, OLUM_Z, OLUM_O, TEMPSUMMON_TIMED_DESPAWN, 3600000);
 
-
-        //We're going to assume pInstance is null - SO DO NOT ACCESS IT.
-        Map *map = m_creature->GetMap();
-
-        //NOTE: We think pInstance is SOMETIMES NULL - therefore this solution DOES NOT USE pInstance.
         if (Map *map = m_creature->GetMap())
         {
-            //Search distance is 999 yards, which is stupid, but it makes sure it will find the object.
             if (GameObject *go = GetClosestGameObjectWithEntry(m_creature, 185117, 999.0f)) 
                 go->RemoveFlag(GAMEOBJECT_FLAGS, GO_FLAG_LOCKED);
         }
+
+        if ((pInstance->GetData(DATA_HYDROSS_EVENT) == SPECIAL) && (pInstance->GetData(DATA_LURKER_EVENT) == SPECIAL) && (pInstance->GetData(DATA_LEOTHERAS_EVENT) == SPECIAL) && (pInstance->GetData(DATA_KARATHRESS_EVENT) == SPECIAL) && (pInstance->GetData(DATA_MOROGRIM_EVENT) == SPECIAL))
+            pInstance->SetData(DATA_UNLOCK_VASHJ_DOOR, SPECIAL);
```

Comment:
Unnecessary to check for DATA_KARATHRESS_EVENT == SPECIAL. If this statement ever get's called then we can assume it's because the boss is dead and infer that DATA_KARATHRESS_EVENT == DONE//SPECIAL.
Same goes for other 5 boss scripts.